json-fleece-core/src/Fleece/Core/Schemas.hs

```diff
@@ -599,39 +401,57 @@ fixedFromScientific name sci =
       then Right (realToFrac normalized)
       else
         Left $
-          "Invalid fixed precision number "
-            <> nameToString name
-            <> ", max precision is "
+          "Invalid fixed precision number. Max precision is "
             <> show (floor (logBase 10 (fromInteger fixedResolution) :: Double) :: Integer)
             <> " decimal places."
 
 string :: Fleece schema => schema String
-string = transform T.pack T.unpack text
+string = transformAnonymous T.pack T.unpack text
 
 utcTime :: Fleece schema => schema Time.UTCTime
 utcTime =
-  iso8601Formatted "UTCTime" ISO8601.iso8601Format AttoTime.utcTime
+  iso8601Formatted $
+    ISO8601Format
+      { iso8601FormatLogical = "date-time"
+      , iso8601FormatString = ISO8601.iso8601Format
+      , iso8601FormatParser = AttoTime.utcTime
+      }
 
 utcTimeWithFormat :: Fleece schema => String -> schema Time.UTCTime
 utcTimeWithFormat = timeWithFormat "UTCTime"
 
 localTime :: Fleece schema => schema Time.LocalTime
 localTime =
-  iso8601Formatted "LocalTime" ISO8601.iso8601Format AttoTime.localTime
+  iso8601Formatted $
+    ISO8601Format
+      { iso8601FormatLogical = "date-time-local"
+      , iso8601FormatString = ISO8601.iso8601Format
+      , iso8601FormatParser = AttoTime.localTime
+      }
 
 localTimeWithFormat :: Fleece schema => String -> schema Time.LocalTime
 localTimeWithFormat = timeWithFormat "LocalTime"
 
 zonedTime :: Fleece schema => schema Time.ZonedTime
 zonedTime =
-  iso8601Formatted "ZonedTime" ISO8601.iso8601Format AttoTime.zonedTime
+  iso8601Formatted $
+    ISO8601Format
+      { iso8601FormatLogical = "date-time"
```

Comment:
Is this "date-time" instead of something like "date-time-zoned" to follow the open-api format specifier?

Reply:
Yes, that's exactly right. I wanted to keep this in line with the OpenAPI spec, which doesn't provide separate formats for times that are solely UTC versus those with useful timezone information. Our own `utcTime` and `zonedTime`  schemas specify how the values will be encoded and decode wrt the information in the Haskell datatypes, we have no way to specify the difference via the OpenAPI `format`